src/app/OperationalSessionSetup.cpp

```diff
@@ -291,6 +295,22 @@ CHIP_ERROR OperationalSessionSetup::EstablishConnection(const ReliableMessagePro
     mCASEClient = mClientPool->Allocate();
     ReturnErrorCodeIf(mCASEClient == nullptr, CHIP_ERROR_NO_MEMORY);
 
+    // TODO: Combine LargePayload flag with DNS-SD advertisements from peer.
+    // Issue #32348.
+    if (mTransportPayloadCapability == TransportPayloadCapability::kLargePayload)
+    {
+#if INET_CONFIG_ENABLE_TCP_ENDPOINT
+        // Set the transport type for carrying large payloads
+        mDeviceAddress.SetTransportType(chip::Transport::Type::kTcp);
+#else
+        ChipLogError(Discovery, "Cannot setup session for large payload. Appropriate Transport unavailable");
+
+        CleanupCASEClient();
```

Comment:
Why aren't we just doing this check before allocating mCASEClient?

Reply:
Moved to before allocation.